### PR TITLE
docs: release process steps

### PR DESCRIPTION
**Pull request by jfatta**
Document the release process.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -37,3 +37,28 @@ If you have to add another go dependency, you can follow the steps:
...
+- update https://github.com/auth0/homebrew-auth0-cli with the latest binary reference
+
+To release a new version:
+
+1. pull the latest changes: 
+   - `$ git checkout main`
+   - `$ git pull origin main`
+2. check the latest tag: 
+   - `$ git fetch`
+   -  `$git tags`
```
Comment (Widcket) [inline CONTRIBUTING.md:56]:
```suggestion
   -  `$ git tags`
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -37,3 +37,28 @@ If you have to add another go dependency, you can follow the steps:
...
+3. create the **new** tag for the new release. For example, if the latest tag is `v0.1.1`, you should create `v.0.1.1`:
```
Comment (Widcket) [inline CONTRIBUTING.md:57]:
```suggestion
3. create the **new** tag for the new release. For example, if the latest tag is `v0.1.1` and you want to release a patch version, you should create `v0.1.2`:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -37,3 +37,28 @@ If you have to add another go dependency, you can follow the steps:
...
+   - `git tag v0.1.1`
```
Comment (Widcket) [inline CONTRIBUTING.md:58]:
```suggestion
   - `git tag v0.1.2`
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -37,3 +37,28 @@ If you have to add another go dependency, you can follow the steps:
...
+4. push the new tag: 
+   - `$ git push origin v.0.1.1`
```
Comment (Widcket) [inline CONTRIBUTING.md:60]:
```suggestion
   - `$ git push origin v.0.1.2`
```

Comment (jfatta) [reply CONTRIBUTING.md:60]: thanks 💤 🤦

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -37,3 +37,27 @@ If you have to add another go dependency, you can follow the steps:
...
+   - `$ git pull origin main`
+2. check the latest tag: 
+   - `$ git fetch`
+   -  `$ git tags`
+3. create the **new** tag for the new release. For example, if the latest tag is `v0.1.1`, you should create `v.0.1.1`:
+   - `git tag v0.1.1`
+4. push the new tag: 
+   - `$ git push origin v0.1.2`
+
+The rest of the process will take place in the github action: https://github.com/auth0/auth0-cli/actions/workflows/goreleaser.yml
```
Comment (Widcket) [inline CONTRIBUTING.md:62]:
```suggestion
The rest of the process will take place in the github action: https://github.com/auth0/auth0-cli/actions/workflows/goreleaser.yml.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -37,3 +37,27 @@ If you have to add another go dependency, you can follow the steps:
...
+Once the workflow finished, a new release will be available for the create tag.
```
Comment (Widcket) [inline CONTRIBUTING.md:63]:
```suggestion
Once the workflow finishes, a new release will be available for the newly created tag.
```

**Review by Widcket — APPROVED** [review]

**Pull request merged**